### PR TITLE
[ABA impact] Show eager loader

### PR DESCRIPTION
**Pull request by W3stside**
Previously in ABA there were some jumping numbers on change.

## Description
This PR adds a more eager loader on typed value change to hide any jumping values.

## Testing:
1. pick token pair
2. change amounts
3. loader is eager

### CODE REVIEW

**Comment by elena-zh** [comment]
Still sometimes can face this issue. See the video from 19 sec:
https://watch.screencastify.com/v/WVE68cGp0pk72VFKQTly

**Comment by W3stside** [comment]
> Still sometimes can face this issue. See the video from 19 sec: https://watch.screencastify.com/v/WVE68cGp0pk72VFKQTly
> 
> Mainnet: Screenshot_6 Screenshot_8

imo this is acceptable.. @gnosis/gp-frontend

**Comment by W3stside** [comment]
@gnosis/gp-qa @gnosis/gp-frontend I am preemptively merging please post merge review

File: src/custom/hooks/usePriceImpact/useQuoteAndSwap.ts [hunk 1]
```diff
@@ -34,6 +35,14 @@ type GetQuoteParams = {
 
 type FeeQuoteParamsWithError = FeeQuoteParams & { error?: QuoteError }
 
+function useOnSwapParamChange<T>({ onChangeCb, predicate }: { predicate: T; onChangeCb: (...params: any[]) => any }) {
+  const { typedValue, INPUT, OUTPUT, independentField } = useSwapState()
+
+  useEffect(() => {
+    Boolean(predicate) && onChangeCb
```
Comment (anxolin) [inline src/custom/hooks/usePriceImpact/useQuoteAndSwap.ts:42]: I kind of get what if does, but i find it weird. If it's a conditional execution of a callback, Also, I find it better to use an IF.
Predicate has type `T` but shouldn't be boolean parameter? Also, more than a predicate seams like a flag which control wether the callback is called when the params change. If that's the case, maybe something more descriptive (`invokeCallbackOnChange: boolean`)

nit: `Boolean(predicate)` we usually do it as !!predicate

File: src/custom/hooks/usePriceImpact/useQuoteAndSwap.ts [hunk 2]
```diff
@@ -48,6 +57,12 @@ export function useCalculateQuote(params: GetQuoteParams) {
   const [quote, setLocalQuote] = useState<QuoteInformationObject | FeeQuoteParamsWithError | undefined>()
   const [loading, setLoading] = useState(false)
 
+  // listens to changed swap params and calls onChangeCb based on predicate Boolean(predicate)
+  useOnSwapParamChange({
+    onChangeCb: () => setLoading(true),
+    predicate: amount,
```
Comment (anxolin) [inline src/custom/hooks/usePriceImpact/useQuoteAndSwap.ts:63]: nit:
`invokeCallbackOnChange: !!amount`

**Review by anxolin — COMMENTED** [review]
APPROVE, Works as expected!

small nitpicks

File: src/custom/hooks/usePriceImpact/useQuoteAndSwap.ts (hunk 2 above)
Comment (W3stside) [reply src/custom/hooks/usePriceImpact/useQuoteAndSwap.ts:63]: (doesnt even need predicate)

**Pull request merged**